src/index.js

```diff
@@ -65,75 +63,58 @@ module.exports = (immutableOpts, done) => {
   opts.out = opts.out || process.cwd();
 
   if (!opts.appPath || typeof opts.appPath !== 'string') {
-    return done(new Error('opts.appPath must be defined as a string'));
+    throw new Error('opts.appPath must be defined as a string');
   }
 
   opts.appPath = path.resolve(process.cwd(), opts.appPath);
 
   if (opts.dmgPath && typeof opts.dmgPath !== 'string') {
-    return done(new Error(`Expected opts.dmgPath to be a string but it was "${typeof opts.dmgPath}"`));
+    throw new Error(`Expected opts.dmgPath to be a string but it was "${typeof opts.dmgPath}"`);
   }
 
   if (!opts.dmgPath && (!opts.out || typeof opts.out !== 'string')) {
-    return done(new Error('Either opts.dmgPath or opts.out must be defined as a string'));
+    throw new Error('Either opts.dmgPath or opts.out must be defined as a string');
   }
 
   if (!opts.dmgPath && (!opts.name || typeof opts.name !== 'string')) {
-    return done(new Error('opts.name must be defined as a string'));
+    throw new Error('opts.name must be defined as a string');
   }
 
   opts.dmgPath = path.resolve(opts.dmgPath || path.join(opts.out, `${opts.name}.dmg`));
 
-  fs.mkdirs(path.dirname(opts.dmgPath), (mkdirErr) => {
-    if (mkdirErr) return done(mkdirErr);
-
-    opts.overwrite = opts.overwrite || false;
```

Comment:
Are we sure appdmg doesn't need this any more?

Reply:
`overwrite` is `electron-installer-dmg` specific.